src/BlockTypes/SingleProduct.php

```diff
@@ -125,6 +125,8 @@ protected function replace_post_for_single_product_inner_block( $block, &$contex
 		if ( $this->single_product_inner_blocks_names ) {
 			$block_name = array_pop( $this->single_product_inner_blocks_names );
 
+			static $global_post_variable_changed;
```

Comment:
Regarding the usage of static in this context, does employing `static` imply that all instances will share the same variable? If that's the case, could it potentially lead to a race condition if we have multiple Single Product blocks added to the page?

Reply:
> does employing static imply that all instances will share the same variable?

Yes: a static variable exists only in a local function scope, but it does not lose its value when program execution leaves this scope. 

> If that's the case, could it potentially lead to a race condition if we have multiple Single Product blocks added to the page?

No, this is a commonly used feature to improve performance in cases such as this one, when we need to persist the variable until we reach the last inner block to reset its post data.

There are multiple examples of its usage across the core of WordPress, but to give an example that is closer to us,`register_block_type_from_metadata` also relies on the same for optimizing blocks registration: https://github.com/WordPress/WordPress/blob/master/wp-includes/blocks.php#L320